parsl/data_provider/data_manager.py

```diff
@@ -24,26 +27,39 @@ def __init__(self, dfk):
         self.dfk = dfk
         self.globus = None
 
-    def stage_in(self, file, executor):
-        """Transport the file from the input source to the executor.
+    def stage_in(self, input, executor):
+        """Transport the input from the input source to the executor, if it is file-like,
+        returning a DataFuture that wraps the stage-in operation.
 
-        This function returns a DataFuture.
+        If no staging in is required - because the `file` parameter is not file-like,
+        then return that parameter unaltered.
 
         Args:
             - self
-            - file (File) : file to stage in
+            - input (Any) : input to stage in. If this is a File or a
+              DataFuture, stage in tasks will be launched with appropriate
+              dependencies. Otherwise, no stage-in will be performed.
             - executor (str) : an executor the file is going to be staged in to.
         """
 
+        if isinstance(input, DataFuture) and input.file_obj.is_remote():
+            file = input.file_obj
+            parent_fut = input
+        elif isinstance(input, File) and input.is_remote():
+            file = input
+            parent_fut = None
+        else:
```

Comment:
Should we have some kind of warning to the user in case they've passed a string path and are expecting a file with that name to be staged?